src/jit/importer.cpp

```diff
@@ -5616,7 +5616,9 @@ var_types           Compiler::impImportCall (OPCODE         opcode,
     // the time being that the callee might be compiled by the other JIT and thus the return
     // value will need to be widened by us (or not widened at all...)
 
-    bool            checkForSmallType = opts.IsJit64Compat();
+    // ReadyToRun code sticks with default calling convention that does not widen small return types.
+
+    bool            checkForSmallType = opts.IsJit64Compat() || opts.IsReadyToRun();
```

Comment:
I think it's a good trade-off which still widens small integers within RyuJit/CoreCLR while it falls back to default calling convention under ReadyToRun, especially for CoreRT use. But this still doesn't fix the issue when we consider an arbitrary altjit like LLILC which doesn't know this special calling convention. So, for correctness, I'd like "checkForSmallType" and its comment to be completely eliminated.
But I'm also okay if all other people agree to keep this model as they are -- using special calling convention for RyuJit/CoreCLR for better performance/code-size while bailing out other alternative Jit model which is mostly seen as an experiment/bring-up.
But I'm also